qualang_tools/plot/plot.py

```diff
@@ -1,17 +1,296 @@
+    plt.tight_layout()
+    return fig
+
+
+def plot_demodulated_data_1d(
+    x: np.ndarray,
+    I: np.ndarray,
+    Q: np.ndarray,
+    x_label: str = None,
+    title: str = None,
+    amp_and_phase: bool = True,
+    fig=None,
+    plot_options: dict = None,
+):
+    """
+    Plot 1D graphs of either 'I' and 'Q' or the corresponding amplitude and phase in a single figure.
+
+    :param x: 1D numpy array representing the x-axis.
+    :param I: 1D numpy array representing the 'I' quadrature.
+    :param Q: 1D numpy array representing the 'Q' quadrature.
+    :param x_label: name for the x-axis label.
+    :param title: title of the plot.
+    :param amp_and_phase: if True, plots the amplitude [np.sqrt(I**2 + Q**2)] and phase [signal.detrend(np.unwrap(np.angle(I + 1j * Q)))] instead of I and Q.
+    :param fig: the matplotlib figure if defined prior to calling the function.
+    :param plot_options: dictionary containing various plotting options ['fontsize', 'color', 'marker', 'linewidth', 'figsize'].
+    :return: the matplotlib figure object.
+    """
+    _plot_options = {
+        "fontsize": 14,
+        "color": "b",
+        "marker": "o",
+        "linewidth": 0,
+        "figsize": None,
+    }
+    if plot_options is not None:
+        for key in plot_options:
+            if key in _plot_options.keys():
+                _plot_options[key] = plot_options[key]
+            else:
+                raise ValueError(
+                    f"The key '{key}' in 'plot_options' doesn't exists. The available options are {[option for option in _plot_options.keys()]}"
+                )
+
+    if amp_and_phase:
+        y1 = np.sqrt(I**2 + Q**2)
+        y2 = signal.detrend(np.unwrap(np.angle(I + 1j * Q)))
+        y1_label = "Amplitude [a.u.]"
+        y2_label = "Phase [rad]"
+    else:
+        y1 = I
+        y2 = Q
+        y1_label = "I [a.u.]"
+        y2_label = "Q [a.u.]"
+
+    if fig is None:
+        fig = plt.figure(figsize=_plot_options["figsize"])
+    plt.suptitle(title, fontsize=_plot_options["fontsize"] + 2)
+    ax1 = plt.subplot(211)
+    plt.cla()
+    plt.plot(
+        x,
+        y1,
+        color=_plot_options["color"],
+        marker=_plot_options["marker"],
+        linewidth=_plot_options["linewidth"],
+    )
+    plt.ylabel(y1_label, fontsize=_plot_options["fontsize"])
+    plt.xticks(fontsize=_plot_options["fontsize"])
+    plt.yticks(fontsize=_plot_options["fontsize"])
+    plt.subplot(212, sharex=ax1)
+    plt.cla()
+    plt.plot(
+        x,
+        y2,
+        color=_plot_options["color"],
+        marker=_plot_options["marker"],
+        linewidth=_plot_options["linewidth"],
+    )
+    plt.xlabel(x_label, fontsize=_plot_options["fontsize"])
+    plt.ylabel(y2_label, fontsize=_plot_options["fontsize"])
+    plt.xticks(fontsize=_plot_options["fontsize"])
+    plt.yticks(fontsize=_plot_options["fontsize"])
+    plt.tight_layout()
+    return fig
+
+
+def plot_channel(ax, data: np.ndarray, title: str):

```

Comment:
I don't get this function, this is the same as just using normal plot, but it decides on the `ylim`?

Reply:
Yes, it was in Lior's lib but I won't add it because it is not very useful and we already have plot 1D and 2D now